ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java

```diff
@@ -118,44 +121,47 @@ private boolean isRestoring() {
     return latestReplayIdx < latestReplay.size();
   }
 
-  private boolean isRecordInLatestReplay(final ConsumerRecord<CommandId, Command> record) {
-    final Pair<CommandId, Command> latestReplayRecord = latestReplay.get(latestReplayIdx);
-    if (record.key().equals(latestReplayRecord.left)
-        && record.value().equals(latestReplayRecord.right)) {
+  private boolean isRecordInLatestReplay(final ConsumerRecord<byte[], byte[]> record) {
+    final Pair<byte[], byte[]> latestReplayRecord = latestReplay.get(latestReplayIdx);
+
+    if (Arrays.equals(record.key(), latestReplayRecord.getLeft())
+        && Arrays.equals(record.value(), latestReplayRecord.getRight())) {
       latestReplayIdx++;
       return true;
     }
 
     return false;
   }
 
-  @Override
-  public void writeRecord(final ConsumerRecord<byte[], byte[]> record) {
-    final ConsumerRecord<CommandId, Command> deserializedRecord;
+  private void throwIfInvalidRecord(final ConsumerRecord<byte[], byte[]> record) {
     try {
-      deserializedRecord = new ConsumerRecord<>(
-          record.topic(),
-          record.partition(),
-          record.offset(),
-          InternalTopicSerdes.deserializer(CommandId.class)
-              .deserialize(record.topic(), record.key()),
-          InternalTopicSerdes.deserializer(Command.class)
-              .deserialize(record.topic(), record.value())
-      );
-    } catch (Exception e) {
-      LOG.error("Failed to deserialize command topic record when backing it up: {}:{}",
-          record.key(), record.value());
-      return;
+      InternalTopicSerdes.deserializer(CommandId.class).deserialize(record.topic(), record.key());
+    } catch (final Exception e) {
+      throw new KsqlException(String.format(
+          "Cannot deserialize record key: %s",
```

Comment:
let's make these error messages a little bit more meaningful (similar to what you head before, mentioning "while backing it up from the replay file") and also include the exception (so that we know why we can't de-serialize it)

Reply:
Done